src/include/nodes/relation.h

```diff
@@ -322,11 +322,11 @@ typedef struct PlannerInfo
 typedef struct CtePlanInfo
 {
 	/*
-	 * A subplan, prepared for sharing among many CTE references by
-	 * prepare_plan_for_sharing(), that implements the CTE. NULL if the
-	 * CTE is not shared among references.
+	 * A subplan prepared for sharing or inlining among many CTE references
+	 * by prepare_plan_for_sharing(), that implements the CTE. NULL if the
+	 * CTE is not shared among references or has not been planned yet.
```

Comment:
Better to write this way:
```
	 * A subplan, that implements the CTE and which is prepared either for
	 * sharing among many CTE references by prepare_plan_for_sharing() or 
	 * for inlining in cases, when sharing produces invalid plans. NULL if
	 * the CTE is not shared among references (gp_cte_sharing is off).
```

Reply:
It can be NULL if it's shared or inlined, but hasn't been planned yet.

Reply:
If we follow this code [branch](https://github.com/arenadata/gpdb/blob/3b03d5eba66bcedc912d09415ef12fb114d75632/src/backend/optimizer/path/allpaths.c#L2034-L2099)  (original inline),  the cteplaninfo->subplan will be always NULL.

Reply:
These points don't cancel each other, so I mentioned both